src/content/blog/javax-jakarta-openrewrite-automigrate/index.md

```diff
@@ -0,0 +1,154 @@
+
+Execute
+
+`./standalone.sh`
+
+to start up the web console, and go to localhost:8080 in your browser. Upload the complete-duke.war file we built previously, click Next
+
+![screenshot-1](./uploadWar.png)
+*screenshot-1: upload the completeDuke.war into Windup*
+
+Select the jakarta-ee target
+
+![screenshot-2](./selecttarget.png)
+*screenshot-2: select the target for the analysis*
+
+Then click through the following pages and eventually select ‘Save and Run’ to start the analysis. When the analysis finishes click on the graph symbol on the right of the analysis list row to load the report:
+
+![screenshot-3](./analysisRow.png)
+*screenshot-3: load the report*
+
+Click on complete-duke.war in the Application List then on the Issues tab to see what needs to be changed in the Mandatory section:
+
+![screenshot-4](./mtaAnalysisReport.png)
+*screenshot-4: view the recommended changes*
+
+We can see the issues are the package name changes, property names, Bootstrapping file names and changes required to XML.
+
+##Auto-Migration to Jakarta EE 9
+
+The CLI flavour of Windup has openrewrite recipes bundled which allow for the auto-migration of an app such as complete-duke to run on a Jakarta EE9-compliant server. Using the windup-cli shell script with the -–openrewrite switch, these recipes can be run to transform the application source code.
+
+Go to the bin folder of your mta-cli distribution and run
+
+```Shell
+./windup-cli --openrewrite "-DactiveRecipes=org.jboss.windup.JavaxToJakarta" "-Drewrite.configLocation=<path-to-mta-cli>/rules/openrewrite/jakarta/javax/imports/rewrite.yml" --input <path-to-local-source-project>/complete-duke/  --goal run
+```
+
+You should see a message like this in the terminal window:
+
+![screenshot-5](./importsTransformRunResults.png)
+*screenshot-5: Transform imports results*
+
+Again from the bin folder, run
+
+```Shell
+./windup-cli --openrewrite "-DactiveRecipes=org.jboss.windup.javax-jakarta.PersistenceXml" "-Drewrite.configLocation=<path-to-mta-cli>/rules/openrewrite/jakarta/javax/xml/rewrite.yml" --input <path-to-local-source-project>/complete-duke/  --goal run
+```
+
+And this time you should see messages like this:
+
+![screenshot-6](./persistenceXmlResults.png)
+*screenshot-6: Transform persistence xml results*
+
+Finally run this from the same location
+
+```Shell
+./windup-cli --openrewrite "-DactiveRecipes=org.jboss.windup.jakarta.javax.BootstrappingFiles" "-Drewrite.configLocation=<path-to-mta-cli>/rules/openrewrite/jakarta/javax/bootstrapping/rewrite.yml" --input <path-to-local-source-project>/complete-duke/  --goal run
+```
+
+Resulting in messages like this:
+
+![screenshot-7](./bootstrappingResults.png)
+*screenshot-7: Transform bootstrapping files results*
+
+If we now run
+
+`mvn clean package`
+
+from the root of the completeDuke source folder, the transformed source project should build correctly without any errors.
+
+#Deployment to Jakarta EE 9 Application Server
```

Comment:
As above,  `# ` (single) is not allowed in the body of the blog post hence the "bigger" heading allowed is `##`